### ISSUE TITLE
Error installing on macOS Mojave

### ISSUE THREAD

**Issue by k06a** [cryptocoinjs/secp256k1-node#135]
>`npm i -g secp256k1`
```
> secp256k1@3.5.2 install /Users/k06a/.npm-packages/lib/node_modules/secp256k1
> npm run rebuild || echo "Secp256k1 bindings compilation fail. Pure JS implementation will be used."


> secp256k1@3.5.2 rebuild /Users/k06a/.npm-packages/lib/node_modules/secp256k1
> node-gyp rebuild

  CXX(target) Release/obj.target/secp256k1/src/addon.o
../src/addon.cc:11:1: error: unknown type name 'secp256k1_context'; did you mean 'secp256k1_context_t'?
secp256k1_context* secp256k1ctx;
^~~~~~~~~~~~~~~~~
secp256k1_context_t
/usr/local/include/secp256k1.h:50:41: note: 'secp256k1_context_t' declared here
typedef struct secp256k1_context_struct secp256k1_context_t;
                                        ^
1 error generated.
make: *** [Release/obj.target/secp256k1/src/addon.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/Cellar/node@8/8.12.0/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:262:23)
gyp ERR! stack     at emitTwo (events.js:126:13)
gyp ERR! stack     at ChildProcess.emit (events.js:214:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
gyp ERR! System Darwin 18.0.0
gyp ERR! command "/usr/local/Cellar/node@8/8.12.0/bin/node" "/usr/local/Cellar/node@8/8.12.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/k06a/.npm-packages/lib/node_modules/secp256k1
gyp ERR! node -v v8.12.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! secp256k1@3.5.2 rebuild: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the secp256k1@3.5.2 rebuild script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/k06a/.npm/_logs/2018-09-25T15_01_46_926Z-debug.log
Secp256k1 bindings compilation fail. Pure JS implementation will be used.
+ secp256k1@3.5.2
added 24 packages from 21 contributors in 4.828s
```

**Comment by k06a** [comment]
Need any solution ASAP :)

**Comment by fanatid** [comment]
I'm not macOS user, if you will find solution this will be great.
If you need any solution then pure js works even if building fail.

**Comment by k06a** [comment]
@fanatid how to install pure js version with npm?

**Comment by fanatid** [comment]
> pure js works even if building fail

if you do not want build bindings you can run `npm install secp256k1 --ignore-scripts`

**Comment by holgerd77** [comment]
Also stumbled into a similar issue on another library, I think this is probably not a library specific issue but some Node.js/node-gyp/Xcode/Mac OS Mojave problem.

Not sure what's going on yet though.

**Comment by aaronisme** [comment]
any progress on this issue?

**Comment by aaronisme** [comment]
find some workaround may be can help others. just delete the file `/usr/local/include/secp256k1.h ` and the npm progress will be succcessed

**Comment by holgerd77** [comment]
(for safety reasons alternative maybe rename or move first)

**Comment by fanatid** [comment]
@aaronisme but from where you have file `/usr/local/include/secp256k1.h`?
maybe if we will move `/usr/local/include` to end of the list this will help? https://github.com/cryptocoinjs/secp256k1-node/blob/25a4b6cb567b49a40f47f50c5ca9a756f5343e4d/binding.gyp#L27

**Comment by aaronisme** [comment]
for me, the file is existed in `/usr/local/include` in my laptop. and yes I renamed the file to another name

**Comment by 7-of-9** [comment]
+1

**Comment by Cengkaruk** [comment]
+1

But no idea where the file come from, not linked anywhere. I thought come from brew packages, but no luck. Closest brew package i get is `libbitcoin` and not linked to the file.

```
$ brew ls -v libbitcoin | grep secp256k1.h
/usr/local/Cellar/libbitcoin/3.5.0_2/libexec/include/secp256k1.h
```

**Comment by aaronisme** [comment]
@fanatid yes, probably it will works

**Comment by fanatid** [comment]
should be fixed now
published in `3.6.0`

**Comment by fanatid** [comment]
for reference, fixed in #137
